### PR TITLE
[sdk/javascript]: unify sign/verify behavior with python

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: javascript does not require canonical signature
solution: add canonical signature check
          port additional edge case tests from python

### CODE REVIEW

File: sdk/javascript/src/symbol/KeyPair.js [hunk 1]
```diff
@@ -40,6 +41,30 @@ class KeyPair {
 	}
 }
 
+const isCanonicalS = encodedS => {
```
Comment (Jaguar0625) [inline sdk/javascript/src/symbol/KeyPair.js:44]: alternatively could promote this and use for both - preference?

**Review by gimre-xymcity — APPROVED** [review]
